src/Database/Statement.php

```diff
@@ -0,0 +1,399 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database;
+
+use Cake\Database\Log\LoggedQuery;
+use Closure;
+use InvalidArgumentException;
+use PDO;
+use PDOStatement;
+use Psr\Log\LoggerInterface;
+use RuntimeException;
+
+class Statement
```

Comment:
```suggestion
class Statement implements Countable, IteratorAggregate
```
`Statement::getInnerIterator()` can return the `\PDOStatement` instance.

Reply:
Yea, I'm missing the iterator implementation here. If we return \PDOStatement, then the iterator won't have any modifiers and will disable buffering.